### PR TITLE
BAH-435 Date picker date in appointment list/calendar view can be cleared with the delete button

### PR DESCRIPTION
**Pull request by smartyrad**
BAH-435 Date picker date in appointment list/calendar view can be cleared with the delete button

### CODE REVIEW

**Comment by smartyrad** [comment]
@mks-d  @sumanmaity112 Can you please review this PR? This is continued from [here](https://github.com/Bahmni/openmrs-module-bahmniapps/pull/142) . I had to create this new PR because I created it from a new branch in my fork.

**Comment by smartyrad** [comment]
As far as the npm package-lock.json file is concerned, according to the documentation, it is mentioned that this file is to be commited as well( I'm not sure whether I must include it in this PR though).

File: ui/app/appointments/directives/datePicker.js [hunk 1]
```diff
@@ -22,18 +23,24 @@ angular.module('bahmni.appointments')
                 $scope.viewDate = $scope.viewDate && dateUtil.addDays($scope.viewDate, 1);
             };
 
-            $scope.$watch("viewDate", $scope.onChange($scope.viewDate));
-
+            $scope.$watch("viewDate", function (viewDate) {
+                $scope.viewDate = viewDate;
+                if (!viewDate) {
```
Comment (sumanmaity112) [inline ui/app/appointments/directives/datePicker.js]: Hi @smartyrad ,
You can simplify the if check. Try to do `$scope.viewDate = viewDate || $scope.lastValidDate`

**Comment by sumanmaity112** [comment]
Hi @smartyrad,
Do you really need `lastValidDate`? If yes, please rename it. `lastValidDate` doesn't seem right name here

**Comment by smartyrad** [comment]
Sure @sumanmaity112. According to you, what must I rename it to? Probably lastSavedDate or savedDate/previousDate?

**Comment by sumanmaity112** [comment]
What about `lastViewDate`? @smartyrad

**Comment by smartyrad** [comment]
@sumanmaity112 Yes, that was discussed with @mks-d on slack as well [here](https://bahmni.slack.com/archives/C03LWT96L/p1522162587000724)

File: ui/app/appointments/directives/datePicker.js (hunk 1 above)
Comment (smartyrad) [reply ui/app/appointments/directives/datePicker.js]: @sumanmaity112 Good suggestion, I have implemented your changes and updated this PR

File: ui/app/appointments/directives/datePicker.js [hunk 2]
```diff
@@ -22,18 +23,22 @@ angular.module('bahmni.appointments')
                 $scope.viewDate = $scope.viewDate && dateUtil.addDays($scope.viewDate, 1);
             };
 
-            $scope.$watch("viewDate", $scope.onChange($scope.viewDate));
-
+            $scope.$watch("viewDate", function (viewDate) {
+                $scope.viewDate = viewDate;
```
Comment (sumanmaity112) [inline ui/app/appointments/directives/datePicker.js]: Hi @smartyrad ,
Please remove this line

Comment (smartyrad) [reply ui/app/appointments/directives/datePicker.js]: @sumanmaity112 Changed that. Sorry for not having noticed it before

**Comment by smartyrad** [comment]
@sumanmaity112 @mks-d Hi, are there any more changes needed ?

**Comment by smartyrad** [comment]
@mks-d True, I thought as much as well. Just updated my commit to exclude the file

**Pull request merged**